openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,743 @@
+                num_par_vals = num_res*6
+                res_values = values[:num_par_vals]
+                cov_values = values[num_par_vals:]
+
+                energy = res_values[0::6]
+                spin = res_values[1::6]
+                gn = res_values[2::6]
+                gg = res_values[3::6]
+                gfa = res_values[4::6]
+                gfb = res_values[5::6]
+
+                for i, E in enumerate(energy):
+                    records.append([energy[i], spin[i], gn[i], gg[i],
+                                    gfa[i], gfb[i]])
+
+                # Build the upper-triangular covariance matrix
+                cov_dim = mpar*num_res
+                cov = np.zeros([cov_dim, cov_dim])
+                indices = np.triu_indices(cov_dim)
+                cov[indices] = cov_values
+
+            # Create pandas DataFrame with resonance data
+            columns = ['energy', 'J', 'neutronWidth', 'captureWidth',
+                       'fissionWidthA', 'fissionWidthB']
+            parameters = pd.DataFrame.from_records(records, columns=columns)
+
+            # Add parameters from File 2
+            parameters = _add_file2_contributions(parameters,
+                                                  resonance.parameters)
+
+        # Compact format - Resonances and individual uncertainties followed by
+        # compact correlations
+        elif lcomp == 2:
+            items, values = endf.get_list_record(file_obj)
+            num_res = items[5]
+            energy = values[0::12]
+            spin = values[1::12]
+            gn = values[2::12]
+            gg = values[3::12]
+            gfa = values[4::12]
+            gfb = values[5::12]
+            par_unc = []
+            for i in range(num_res):
+                res_unc = values[i*12+6 : i*12+12]
+                # Delete 0 values (not provided in evaluation)
+                res_unc = [x for x in res_unc if x != 0.0]
+                par_unc.extend(res_unc)
+
+            records = []
+            for i, E in enumerate(energy):
+                records.append([energy[i], spin[i], gn[i], gg[i],
+                                gfa[i], gfb[i]])
+
+            corr = endf.get_intg_record(file_obj)
+            cov = np.diag(par_unc).dot(corr).dot(np.diag(par_unc))
+
+            # Create pandas DataFrame with resonacne data
+            columns = ['energy', 'J', 'neutronWidth', 'captureWidth',
+                       'fissionWidthA', 'fissionWidthB']
+            parameters = pd.DataFrame.from_records(records, columns=columns)
+
+            # Determine mpar (number of parameters for each resonance in
+            # covariance matrix)
+            nparams, params = parameters.shape
+            covsize = cov.shape[0]
+            mpar = int(covsize/nparams)
+
+            # Add parameters from File 2
+            parameters = _add_file2_contributions(parameters,
+                                                  resonance.parameters)
```

Comment:
Same comment here -- the lines starting from `columns = ...` looks redundant on the branches of the if.